public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -467,6 +414,110 @@ export const RoleMenuOption = React.forwardRef<HTMLDivElement, React.PropsWithCh
 
 RoleMenuOption.displayName = 'RoleMenuOption';
 
+interface RoleMenuGroupsSectionProps {
+  roles: Role[];
+  renderedName: string;
+  menuSectionStyle: string;
+  groupHeaderStyle: string;
+  optionBodyStyle: string;
+  showGroups?: boolean;
+  optionGroups: Array<{
+    name: string;
+    options: Role[];
+    value: string;
+  }>;
+  onChange: (value: string) => void;
+  onOpenSubMenuRMGS: (value: string) => void;
+  onCloseSubMenu?: (value: string) => void;
+  groupSelected: (group: string) => boolean;
+  groupPartiallySelected: (group: string) => boolean;
+  disabled?: boolean;
+  subMenuNode?: HTMLDivElement;
+  showSubMenu: boolean;
+  openedMenuGroup: string;
+  subMenuOptions: Role[];
+  selectedOptions: Role[];
+  onChangeSubMenu: (option: Role) => void;
+  onClearSubMenu: () => void;
+  showOnLeftSubMenu: boolean;
+}
+
+export const RoleMenuGroupsSection = React.forwardRef<HTMLDivElement, RoleMenuGroupsSectionProps>(
```

Comment:
Ideally all these components should be in their own file to make it easier reading the code.

Reply:
Agreed. I decided to not rework it for now since that's outside of scope.